[IV-25-26] Objetivo 5

gabrielherreraloz opened the pull request:
* [x] ¿Se han establecido criterios previos de elección de la imagen base, así
como los criterios de búsqueda que han conducido a las imágenes que se están
considerando?
* [x] ¿Se ha tenido en cuenta que la imagen es para testear, y que por tanto
tiene una función específica, tal como testear cómo va nuestra aplicación en la
última versión publicada del lenguage?
* [x] ¿Se ha tenido en cuenta que la versión de la imagen base sea la última?
* [x] ¿Se han tenido en cuenta criterios reales para la elección, y no criterios
irrelevantes como cuanto tiempo se tarda en construirla? (Se va a construir una
vez, se va a usar cientos o miles de veces)
* [x] ¿Se han seguido las buenas prácticas en las órdenes que construyen
   la imagen? ¿Especialmente las relativas a usuarios no privilegiados y demás?
* [x] ¿La imagen incluye sólo y exclusivamente la infraestructura
   necesaria para pasar los tests? Por ejemplo, si se ha elegido la imagen
   "oficial" (lo que hay que hacer sólo en casos muy justificados) ¿se ha
   comprobado que lo que se instale sea sólo y exclusivamente lo necesario para
   nuestro proyecto?
* [x] ¿Se han documentado y enlazado los commits a las imágenes que se
   han testeado?
* [x] ¿Está establecido correctamente como `WORKDIR` el que se va a usar en los
   tests que se van a lanzar?
* [x] ¿Se usa algún directorio adecuado para copiar contenidos del repositorio
en vez del directorio de trabajo?
* [x] ¿Se han establecido claramente los criterios de búsqueda y de aceptación de
   la imagen base?
* [x] ¿Tienes claro cuales son las condiciones en las que tiene que actualizarse
la imagen en DockerHub y has configurado el workflow de acuerdo con ello? En
particular, ¿sabes qué condiciones tienen que darse para que se actualice y qué
condiciones *no* deben darse?
* [x] Si se usan GitHub actions para su publicación, ¿se han considerado
cuidadosamene los criterios (ramas, paths) para que se lancen?
* [x] ¿Se ha comprobado que el ENTRYPOINT del contenedor no escriba nada en el
directorio que monta, ya que no tiene derecho de escritura?
* [x] ¿Se ha abierto un shell en el contenedor una vez construida la imagen y
comprobado que los ficheros que se necesitan están todos incluidos, y que no hay
ficheros que no se necesiten?

gabrielherreraloz committed:
Creada base del Objetivo

Se ha creado el dockerfile básico, y un archivo donde es especifica el nombre de usuario en docker.

gabrielherreraloz committed:
Creada documentacion de las imágenes docker(#27)

Se ha documentado en un arcivo.md las decisiones y criterios tomados en la elección de las imágenes para la contenerización del proyecto, detallando cada paso e incliyendo en cada uno la documentación oficial asociada.

Añadido enlace a documentación en README(#27)

Modificada documentación sobre Multi-Stage(#27)

Se ha suprimido toda la parte de multi-stage y se ha cambiado de estrategia al uso de una única imágen, ya que con el comando seleccionado para las pruebas, no funcionará este enfoque.

gabrielherreraloz committed:
Creado archivo de contenerización(#27)

Se ha creado el dockerfile correspondiente a parti de la imagen base de alpine seleccionada en la documentación, en la que se instalan make y gcc para la ejecución, y además se crea el directorio correspondiente para el comando utilizado para lanzar el contenedor, también se otorga permisos al usuario 1001 para su correcto funcionamiento y finalmente se lanza la ejecución de tests.
Además, se ha solucionado el makefile incorrecto, añadiendo su apartado corrrespondiente de ejecuión de tests.

gabrielherreraloz committed:
Creada githubaction(#27)

Se ha creado una github actión para lanzar el dockerfile utilizando la base del action Docker Publisher predeternminada de github, adaptandola al caso de este proyecto.

gabrielherreraloz committed:
FIX-githubaction(#27)

Se ha eliminado el guardado en cache de github por parte de la githubaction ya que producia errores.

gabrielherreraloz commented:
@GabrielFranciscoSM revisión porfa

GabrielFranciscoSM commented on Dockerfile:
```diff
@@ -0,0 +1,17 @@
+RUN mkdir -p /tmp/go-cache && chmod -R 777 /tmp/go-cache
+ENV GOCACHE=/tmp/go-cache
+
+COPY . .
```
¿Con este comando estas copiando sólo los archivos necesarios?

GabrielFranciscoSM commented on Dockerfile:
```diff
@@ -0,0 +1,17 @@
+WORKDIR /app/test
+
+# Creamos la carpeta de cache y le damos permisos al usuario 1001
+RUN mkdir -p /tmp/go-cache && chmod -R 777 /tmp/go-cache
```
¿Es necesario dar todos los privilegios en este directorio (777)?

gabrielherreraloz replied:
Tienes razón, sería más correcto y seguro dar los permisos únicamente al usuario1001

JJ replied:
Tienes que resolver todos los comentarios.  Y no, no puedes dar privilegios solo a ese usuario porque no sabes si se va a ejecutar el contenedor con ese usuario.
Lo correcto en todo caso es dar los privilegios que sean necesarios a quien sea necesario. Por ejemplo, ¿para qué das privilegios a grupo?

GabrielFranciscoSM reviewed (commented):
Te he dejado un par de comentarios sobre el dockerfile.

gabrielherreraloz committed:
Eliminados archivos de objetivo

Se han eliminado el dockerfile y la documentación ya que eran conecptualmente erroneas.

Copiado de archivos en dockerfile(#27)

Se ha modificado el copiado de los archivos de dockerfile para copiar unicamente los archivos indispensables para la ejecución de tests

REstaurados permisos en Dockerfile(#27)

Se han vuelto a otorgar permisos a todos los usuarios desde el dockerfile, decision tomada en conjunto con los comentarios de @GabrielFranciscoSM

Eliminadas versiones en documentación(#27)

Se ha eliminado la especificación de la versión elegida en la documentación de la elección de imagen base para el contenedor.

Modificada documentación scratch(#27)

Se ha modificado la documentación sobre la decición de descartar scratch como imagen base puesto que la justificación de no incluir shell, no era aplicable al contexto del PMV p`resentado.`

Eliminado Dockerfile

Se ha eliminado para empezarlos desde 0 por inconsistencia de los issues

Otorgados permisos a grupo(#27)

Al quitar los permisos de grupo, aunque en teoria no son necesarios en este contexto, falla la ejecución de permisos si no se otorgan pues al detectar linux un 0 en grupos, lo asume para todos los usuarios.

Eliminado gcc en dockerfile(#27)

Se ha eliminado la instalación de gcc del contenedor ateniendonos al PMV y al tamaño mínimo del contenedor creado, se creía que era imprescindible la instalación de gcc para hacer funcionar make con go pero no es necesario en este contexto, lo cual, además reduce el almacenamiento utilizado,

Eliminada especificación de versión(#27)

Se ha eliminado la especificación de la versión de la imágen base, la forma en la que está escrito ahora utiliza la última versión estable

Especificada elección imagen base(#27)

Se ha especificado en la documentación que se ha elegido Alpine en su variante para Go.

Eliminado Dockerfile

Se ha eliminado para empezarlos desde 0 por inconsistencia de los issues

Creado Dockerfile(#27)

Se ha creado un archivo Dockerfile a partir de la imagen sleeccionada de Alpine para Go-lang.
Se ha instalado la herramienta make.
Se han otorgado permisos de superusuario para que Go pueda escribir en cache al compilar con 777 pero se le ha añadido un 1 delante quedando como 1777, lo cual permite a todos los usuarios escribir y crear nuevos archivos que pertenezcan a él mismo, es decir, no permirte borrar o modificar archivos que ya había creados antes, pero permiote a los ususarios crear nuevo y modificarlos.
Finalmente, se ejecutan los tests a través de make.

Eliminado criterio estabilidad(#27)

Se ha eliminado el criterio de seleccionar una imagen base estable, ya que se presupone que se utilizará la ultima versión estable de la seleccionada.

Cambiado especificacion de version

Se ha cambiado la version del login de docker en la githubaction en lugar del commit por directamente el nombre de version para hacer mas sencilla su lectura y comprension.

Eliminado Dockerfile

Creación entorno de ejecución básico(#27)

Se ha creado la base del archivo Dockerfile, en el cual se instalan las dependencias necesarias y, además se define un usuario sin privilegios para no lanzar la ejecución de los test utilizando root por omisión
También se selecciona como directorio principal de trabajo /app/test, situando esta orden al final del Dockerfile ya que las ordenes previas a estas, se lanzan desde rutas más globales, mientras que las que se situan porteriores a esta, como el ENTRYPOINT, se ejecutan desde el directorio especificado.

Configurado GOCACHE y permisos(#28)

Puesto que el compilador de Go necesita un directorio sobre el que escribir para caché, utilizando el usuario root, se crea el directorio /tmp/go-cache para este cometido, y además se especifican los permisos asociados a este directorio.
Estos cambios sobre los permisos se han realizado para facilitar al compilador de Go la escritura y uso de GOCACHE  pero sin comprometer el código y carpetas que ya existían previamente, lo cual resuelve el problema existente, y no genera nuevos problemas de permisos y seguridad.

Para ello se especifica 777, que otorga permisos al dueño del directorio (primer 7), al grupo al que pertenece el dueño (segundo 7) y a cualquier otro usuario fuera del grupo del dueño (tercer 7).
El 7 especifica permisos de lectura, escritura y ejecución.
Además, se ha añadido un 1 delante, llamado stickybit, el cual restringe los permisos comentados a que no pueda eliminarse contenido por usuarios los cuales no sean propietarios del mismo.

FIX- Atomizada orden permisos

Se ha reducido la orden de la creación y los permisos de la carpeta de GOCACHE para hacerlo en un solo paso más atómico, creando la carpeta con los permisos desde su creación.

Archivos de objetivo eliminados

gabrielherreraloz commented:
@GabrielFranciscoSM listo de nuevo

GabrielFranciscoSM commented on Dockerfile:
```diff
@@ -0,0 +1,29 @@
+WORKDIR /app/test
+
+# Creamos la carpeta de cache y le damos permisos al usuario 1001
+RUN mkdir -p /tmp/go-cache && chown -R 1001:1001 /tmp/go-cache
```
¿Es correcto fijar sólo un UID? No podemos cercionarnos de que gihub actions o cualquier plataforma CI/CD use un UID fijo. En el comentario anterior me refería a si es necesario dar permisos de lectura, escritura y ejecución generales (777). No sé como funciona el caché con go, por eso te lo comentaba.

gabrielherreraloz replied:
si tienes razón... lo he hecho con el 1001 principalmente porque es con el que se hacen las pruebas del objetivo, pero tienes razón.

gabrielherreraloz replied:
> ¿Es correcto fijar sólo un UID? No podemos cercionarnos de que gihub actions o cualquier plataforma CI/CD use un UID fijo. En el comentario anterior me refería a si es necesario dar permisos de lectura, escritura y ejecución generales (777). No sé como funciona el caché con go, por eso te lo comentaba.

Como solución volvería a poner 777 para dar permisos generales a todos los usuarios ya que en Go se necesita permisos de escritura sobre cache para compilar y para los tests. Sin eso no va a funcionar, y como es solo ejecución de tests, me parece un mal menor. ¿Qué opinas?

GabrielFranciscoSM replied:
Vale, no lo veo mal

GabrielFranciscoSM reviewed (approved):
Lo veo bien salvo el comentario que te he puesto.

gabrielherreraloz commented:
@jj listo para revision

JJ commented on docs/Contenedor.md:
```diff
@@ -0,0 +1,31 @@
+  Cuenta con un tamaño mucho menor que la versión de Debian, pesando aproximadamente unos 64,5MB de forma comprimida, llegando a pesar 329MB al expandirse en el disco, por otro lado, no incluye Make instalada por defecto, aunque puede instalarse en un paso posterior sin apenas aumentar el uso de almacenamiento, ni el tiempo de lanzamiento del contenedor.
+
+  **No ofical: Scratch**: [Documentación](https://hub.docker.com/_/scratch)
+  Se trata de una imagen vacía sin nada instalado, es la opción más ligera y segura pero esto dificulta mucho su uso al no tener instalado ni siquiera shell.
```
¿Realmente necesitas el shell para hacer test en go?

JJ commented on docs/Contenedor.md:
```diff
@@ -0,0 +1,31 @@
+Finalmente se ha seleccionado Alpine (golang:1.25.5-alpine3.23) debido a su reducido tamaño con respecto a las otras opciones tenidas en cuenta, además, aunque no cuente con make instalado por defecto, su instalación en su entorno es muy sencilla y ligera.
+
+## Resumen de elecciones
+Imagen base seleccionada: Alpine (golang:1.25.5-alpine3.23)
```
Ya os advertí que no debíais elegir *nunca* versiones específicas de *nada*. Se trata de una imagen para test.

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,29 @@
+ENV GOCACHE=/tmp/go-cache
+
+# Copiado de dependencias
+COPY go.mod ./
```
¿Con qué privilegios estás ejecutando estas copias?

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,29 @@
+
+# Copiado de las carpetas de código
+COPY internal/ ./internal/
+COPY pkg/ ./pkg/
```
Las carpetas de código *no se copian*. El código que se testea es externo a la imagen. ¿para qué vas a hacer una imagen para tstear siempre el mismo código?

JJ reviewed (commented):
Me parece que no has acabado de entender qué se busca con esta imagen docker. Por favor, dale otra vuelta.

gabrielherreraloz commented:
@JJ listo, ya he implementado los cambios, estaba enfocando el dockerfile a problemas futuros sin centrarme en el PMV que ahora es unicamente ejecutar tests.

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,14 @@
+FROM golang:1.25.5-alpine3.23
```
Esta no es la imagen seleccionada arriba.

gabrielherreraloz replied:
¿Como que no? ¿Por haber puesto la versión te refieres?

JJ replied:
Claro. ¿Por qué si no?

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,14 @@
+FROM golang:1.25.5-alpine3.23
+
+# Es necesario instalar make y gcc para que "make test" funcione
+RUN apk add --no-cache make gcc musl-dev
```
¿Por qué hacen falta?

JJ reviewed (commented):
Aparte de lo dicho, ¿con qué usuario y privilegios se está ejecutando make test?

gabrielherreraloz commented:
@JJ He eliminado la especificación de versión de la imagen base, he vuelto a otorgar permisos de grupo ya que si no, devolvía error de permisos al ejecutar los tests, he eliminado gcc y musl-dev porque finalmente no son necesarios para este proytecto y ahorran espacio de la imagen final.

Respondiendo a la pregunta ¿con qué usuario y privilegios se está ejecutando make test?:
Al ejecutar "docker run --rm -u 1001 -v ${PWD}:/app/test gabrielherreraloz15/iv-ghl"
make test se ejecuta con el usuario 1001, por lo que es necesario darle permisos con 1777, que da permisos de escritura, pero los restringe para que no pueda eliminar archivos a su gusto y así no dar libertad total.

JJ commented:
> @JJ He eliminado la especificación de versión de la imagen base, he vuelto a otorgar permisos de grupo ya que si no, devolvía error de permisos al ejecutar los tests, he eliminado gcc y musl-dev porque finalmente no son necesarios para este proytecto y ahorran espacio de la imagen final.

Si no has creado un issue, no sirve de nada esta explicación.
> 
> Respondiendo a la pregunta ¿con qué usuario y privilegios se está ejecutando make test?: Al ejecutar "docker run --rm -u 1001 -v ${PWD}:/app/test gabrielherreraloz15/iv-ghl" make test se ejecuta con el usuario 1001, por lo que es necesario darle permisos con 1777, que da permisos de escritura, pero los restringe para que no pueda eliminar archivos a su gusto y así no dar libertad total.

No te estoy preguntando eso. Te estoy preguntando qué ocurre si no pones `-u 1001`
También convendría que miraras y siguieras las mejores prácticas de creación de una imagen docker.

gabrielherreraloz commented:
> > @JJ He eliminado la especificación de versión de la imagen base, he vuelto a otorgar permisos de grupo ya que si no, devolvía error de permisos al ejecutar los tests, he eliminado gcc y musl-dev porque finalmente no son necesarios para este proytecto y ahorran espacio de la imagen final.
> 
> Si no has creado un issue, no sirve de nada esta explicación.
> 
> > Respondiendo a la pregunta ¿con qué usuario y privilegios se está ejecutando make test?: Al ejecutar "docker run --rm -u 1001 -v ${PWD}:/app/test gabrielherreraloz15/iv-ghl" make test se ejecuta con el usuario 1001, por lo que es necesario darle permisos con 1777, que da permisos de escritura, pero los restringe para que no pueda eliminar archivos a su gusto y así no dar libertad total.
> 
> No te estoy preguntando eso. Te estoy preguntando qué ocurre si no pones `-u 1001` También convendría que miraras y siguieras las mejores prácticas de creación de una imagen docker.

si no pones -u 1001 se ejecuta como root no? con todos los permisos

gabrielherreraloz commented:
> > @JJ He eliminado la especificación de versión de la imagen base, he vuelto a otorgar permisos de grupo ya que si no, devolvía error de permisos al ejecutar los tests, he eliminado gcc y musl-dev porque finalmente no son necesarios para este proytecto y ahorran espacio de la imagen final.
> 
> Si no has creado un issue, no sirve de nada esta explicación.
> 
> > Respondiendo a la pregunta ¿con qué usuario y privilegios se está ejecutando make test?: Al ejecutar "docker run --rm -u 1001 -v ${PWD}:/app/test gabrielherreraloz15/iv-ghl" make test se ejecuta con el usuario 1001, por lo que es necesario darle permisos con 1777, que da permisos de escritura, pero los restringe para que no pueda eliminar archivos a su gusto y así no dar libertad total.
> 
> No te estoy preguntando eso. Te estoy preguntando qué ocurre si no pones `-u 1001` También convendría que miraras y siguieras las mejores prácticas de creación de una imagen docker.

Esque no entiendo de que debo crear el issue, ya tengo un issue creado donde especifico dentro un par de cosas: #27 
además, no solo lo explico aquí, lo he dejado todo tambien explicado en los mensajes de commit

JJ commented:
> > > @JJ He eliminado la especificación de versión de la imagen base, he vuelto a otorgar permisos de grupo ya que si no, devolvía error de permisos al ejecutar los tests, he eliminado gcc y musl-dev porque finalmente no son necesarios para este proytecto y ahorran espacio de la imagen final.
> > 
> > 
> > Si no has creado un issue, no sirve de nada esta explicación.
> > > Respondiendo a la pregunta ¿con qué usuario y privilegios se está ejecutando make test?: Al ejecutar "docker run --rm -u 1001 -v ${PWD}:/app/test gabrielherreraloz15/iv-ghl" make test se ejecuta con el usuario 1001, por lo que es necesario darle permisos con 1777, que da permisos de escritura, pero los restringe para que no pueda eliminar archivos a su gusto y así no dar libertad total.
> > 
> > 
> > No te estoy preguntando eso. Te estoy preguntando qué ocurre si no pones `-u 1001` También convendría que miraras y siguieras las mejores prácticas de creación de una imagen docker.
> 
> Esque no entiendo de que debo crear el issue, ya tengo un issue creado donde especifico dentro un par de cosas: #27 además, no solo lo explico aquí, lo he dejado todo tambien explicado en los mensajes de commit

De veras, Gabriel, dale un poco una vuelta a lo que te voy diciendo. Primero, el issue es por que has incluido cosas en código (el Dockerfile) que no están justificadas. Mira el principio de la conversación. También en un Dockerfile hay que incluir código sólo si resuelve problemas, y si no pones claro el problema en un issue *el código es erróneo*, por lo que el Dockerfile estaría mal. Lo que se aplica en el objetivo 2 se aplica también aquí. Los mensajes de commit deben justificar las decisiones tomadas para resolver un issue, no ser el mismo issue.

Finalmente, lo de las mejores prácticas deberías de leértelo en general, no solo por esto, sino en general. Y ser consciente en todo momento de cómo se va a ejecutar cada orden, qué permisos tiene y qué pasa si lo estás ejecutando, por omisión, con privilegios de superusuario.

Decir que "no entiendes algo" no ayuda en absoluto a que lo entiendas. Trata de mirar las fuentes que se te dicen, y *luego* dí específicamente qué es lo que no entiendes de esas fuentes. Si no miras las fuentes, ni haces la práctica habitual desde el objetivo 2, no vas a solucionar el problema en absoluto ni avanzar hacia la consecución del objetivo.

gabrielherreraloz commented:
> > > > @JJ He eliminado la especificación de versión de la imagen base, he vuelto a otorgar permisos de grupo ya que si no, devolvía error de permisos al ejecutar los tests, he eliminado gcc y musl-dev porque finalmente no son necesarios para este proytecto y ahorran espacio de la imagen final.
> > > 
> > > 
> > > Si no has creado un issue, no sirve de nada esta explicación.
> > > > Respondiendo a la pregunta ¿con qué usuario y privilegios se está ejecutando make test?: Al ejecutar "docker run --rm -u 1001 -v ${PWD}:/app/test gabrielherreraloz15/iv-ghl" make test se ejecuta con el usuario 1001, por lo que es necesario darle permisos con 1777, que da permisos de escritura, pero los restringe para que no pueda eliminar archivos a su gusto y así no dar libertad total.
> > > 
> > > 
> > > No te estoy preguntando eso. Te estoy preguntando qué ocurre si no pones `-u 1001` También convendría que miraras y siguieras las mejores prácticas de creación de una imagen docker.
> > 
> > 
> > Esque no entiendo de que debo crear el issue, ya tengo un issue creado donde especifico dentro un par de cosas: #27 además, no solo lo explico aquí, lo he dejado todo tambien explicado en los mensajes de commit
> 
> De veras, Gabriel, dale un poco una vuelta a lo que te voy diciendo. Primero, el issue es por que has incluido cosas en código (el Dockerfile) que no están justificadas. Mira el principio de la conversación. También en un Dockerfile hay que incluir código sólo si resuelve problemas, y si no pones claro el problema en un issue _el código es erróneo_, por lo que el Dockerfile estaría mal. Lo que se aplica en el objetivo 2 se aplica también aquí. Los mensajes de commit deben justificar las decisiones tomadas para resolver un issue, no ser el mismo issue.
> 
> Finalmente, lo de las mejores prácticas deberías de leértelo en general, no solo por esto, sino en general. Y ser consciente en todo momento de cómo se va a ejecutar cada orden, qué permisos tiene y qué pasa si lo estás ejecutando, por omisión, con privilegios de superusuario.
> 
> Decir que "no entiendes algo" no ayuda en absoluto a que lo entiendas. Trata de mirar las fuentes que se te dicen, y _luego_ dí específicamente qué es lo que no entiendes de esas fuentes. Si no miras las fuentes, ni haces la práctica habitual desde el objetivo 2, no vas a solucionar el problema en absoluto ni avanzar hacia la consecución del objetivo.

De acuerdo, voy a borrar la entrega del dockerfile y lo vuelvo a subir cuando ponga en orden los issues.

gabrielherreraloz commented:
@GabrielFranciscoSM echale un vistazo de nuevo porfa, sobre todo la asociación con el issue que hay.

GabrielFranciscoSM commented on .github/workflows/docker-publish.yml:
```diff
@@ -0,0 +1,33 @@
+
+      - name: Log into dockerhub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
```
¿Por qué fijas un commit en específico y no la versión?

gabrielherreraloz replied:
Porque lo cogí del documento base que crea github, y como es lo mismo pues lo deje asi.

GabrielFranciscoSM commented on docs/Contenedor.md:
```diff
@@ -0,0 +1,31 @@
+
+  **Soporte de gestor de tareas**: Capacidad para instalar y ejecutar make, el gestor de tareas de este proyecto.
+
+  **Estabilidad**: Debe seleccionarse una versión estable para garantizar su correcto funcionamiento, quedan descartadas versiones en fase testing o en RC(Realease candidate). 
```
Si la estabilidad es un criterio, ¿Por qué no se nombra en el resto del documento?

gabrielherreraloz replied:
Porque lo tomaba cuando utilizaba las versiones, lo quito entonces.

GabrielFranciscoSM reviewed (commented):
En cuanto al Issue, ¿el título es coherente con el problema que solucionas y con el cuerpo del propio Issue? ¿Es un entorno de ejecución de todo el proyecto el problema a plantear en este Issue?

gabrielherreraloz commented:
@GabrielFranciscoSM he realizado algunos cambios, echale un vistazo cuando puedas

GabrielFranciscoSM reviewed (approved):
Lo que hay lo veo bien.

gabrielherreraloz commented:
@JJ Listo de nuevo

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,10 @@
+
+WORKDIR /app/test
+
+RUN mkdir -p /tmp/go-cache && chmod 1777 /tmp/go-cache
```
Sigues sin justificar por qué estás específicamente usando estos privilegios para ese directorio.

gabrielherreraloz replied:
Como que no lo justifico? Si en el mensaje de commit dice:
"para que Go pueda escribir en cache al compilar con 777 pero se le ha añadido un 1 delante quedando como 1777, lo cual permite a todos los usuarios escribir y crear nuevos archivos que pertenezcan a él mismo, es decir, no permirte borrar o modificar archivos que ya había creados antes, pero permiote a los ususarios crear nuevo y modificarlos."

JJ replied:
Vamos a ver, Gabriel. Estoy de vacaciones y te rogaría que hicieras un pequeño esfuerzo por entender lo que te estoy diciendo, al menos parejo al que estoy haciendo yo corrigiéndotelo ya DOS veces, las dos durante vacaciones.
1. El mensaje de commit sigue diciendo "creado Dockerfile" algo que ya dijimos hace meses que *no* era la mejor forma de hacerlo.
2. Sigue apuntando al issue #27, que dice "no hay un entorno de ejecución de tests". No hay nada *específico* que justifique este código.
3. Lo que has escrito tú o lo que lo haya escrito no dice qué decisión se ha tomado, simplemente describe *mal* lo que hace ese comando. Yo sé lo que hace ese comando y tú posiblemente no lo sabes, pero eso no importa, lo que tienes que justificar es *porqué* pone un 1 y tres 7s y no 755 o 757 o 644 o =755. 
4. En general, volver a poner lo mismo que se ha comentado antes y/o discutir lo que se ha comentado no conduce a que alcances el objetivo más rápido. Decir "no lo entiendo" tampoco.

Así que, por favor
1. Expón claramente en un issue qué problemas hay con la creación de esta imagen Docker para Go
2. Crea comandos que solucionen ese problema específico, y di en el mensaje de commit por qué has tomado esa decisión y no alguna de las muchas otras posibles

Esto es algo que venimos haciendo desde el objetivo 2, no entiendo que ahora empecemos a sorprendernos.

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,10 @@
+
+RUN apk add --no-cache make
+
+WORKDIR /app/test
```
¿Por qué pones esta orden precisamente aquí? ¿La orden siguiente lo va a usar?

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,10 @@
+RUN mkdir -p /tmp/go-cache && chmod 1777 /tmp/go-cache
+ENV GOCACHE=/tmp/go-cache
+
+ENTRYPOINT ["make", "test"]
```
Te vuelvo a repetir lo mismo: ¿con qué usuario o privilegios estás ejecutando, por omisión, esta orden?

JJ reviewed (commented):
Salvo algún cambio puntual, el fichero Docker sigue teniendo el mismo problema, que en general es
1. No creas código en él para resolver problemas, ni esos problemas están en un issue. El commit dice "creado Dockerfile"
2. El orden de los comandos en el mismo no evidencia que tengas claro, en cada momento, cuales son los privilegios con que se ejecutan, o donde se ejecutan.

gabrielherreraloz commented:
@GabrielFranciscoSM he subido una nueva versión con algunas correcciones. Me gustaría que revisaras sobre todo el nuevo issue que he creado y los dos commits realizados, a ver que opinas de los issues que asociado y la justificación del código realizado. Gracias.

GabrielFranciscoSM commented:
Creo que tienes algún comentario sin resolver, aunque lo hallas justificado mejor es conveniente resolverlos de alguna manera.

GabrielFranciscoSM reviewed (approved):
Lo apruebo. Creo que reflejas adecuadamente la motivación de las decisiones tomadas en los commits, que es, como dice @JJ , la razón de ser de los mensajes.

gabrielherreraloz commented:
@JJ listo

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,17 @@
+FROM golang:alpine
```
Llevamos tres iteraciones (o más) de esto y aparentemente [no has leído el README de la imagen docker y lo que dice de Alpine](https://hub.docker.com/_/golang).

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,17 @@
+
+RUN apk add --no-cache make
+
+RUN addgroup -S test_gp && adduser -S test_usr -G test_gp
```
¿Un producto mínimo requiere que añadas un grupo?

JJ commented on docs/Contenedor.md:
```diff
@@ -0,0 +1,29 @@
+  **Oficial: Basada en Alpine**: [Repositorio](https://github.com/docker-library/golang/blob/adf0ff9a10a766d7dc1b394d52544f3a1b1da4e2/1.25/alpine3.23/Dockerfile)
+  Cuenta con un tamaño mucho menor que la versión de Debian, pesando aproximadamente unos 64,5MB de forma comprimida, llegando a pesar 329MB al expandirse en el disco, por otro lado, no incluye Make instalada por defecto, aunque puede instalarse en un paso posterior sin apenas aumentar el uso de almacenamiento, ni el tiempo de lanzamiento del contenedor.
+
+  **No ofical: Scratch**: [Documentación](https://hub.docker.com/_/scratch)
```
Aparte de que se dice "oficial", esta es una imagen totalmente oficial, pero que se usa *para otra cosa diferente*. Es más, en Go se suele usar una contrucción de imagen *en dos fases* y esta puede ser útil para una de las fases. Deberíais informaros un poco más sobre lo que ponéis.

JJ reviewed (commented):
Aparte de lo comentado, no has hecho un esfuerzo mínimo en buscar imágenes de empresas que se dedican a este menester, ni de examinar la posibilidad de crear tú mismo una imagena a partir de una imagen base. Y llevamos ya tres iteraciones, o más si tenemos en cuenta las de tus compañeros.
Yo entiendo que a estas alturas estéis hartos del cuatrimestre, de la carrera, y por supuesto de IV. Pero únicamente os estoy pidiendo *lo que ya os he pedido en el resto de los objetivos y que hicisteis a la perfección*. Por haber dejado las clases no vamos a dejar de seguir ese espíritu. Os ruego, por favor, que lo hagáis así que ya os queda muy poco.

JJ commented:
Has vuelto a eliminar el Dockerfile para eliminar los comentarios por tercera vez. No es una buena idea, y te rogaría que no lo volvieras a hacer. Las dos veces anteriores no han resultado en un acercamiento al objetivo sino, al contrario, has vuesto a enviar el mismo objetivo con justificaciones diferentes, lo que es un error.
Te rogaría que hicieras cambios *puntuales* en las líneas indicadas. Te recuerdo también que a partir de la tercera revisión (y llevas ya 4) solo se hacen revisiones adicionales si hay alguna mejora; si no hay mejora se puede fallar el objetivo en la convocatoria en la que se esté enviando, que es la ordinaria. Así que te rogaría que hicieras esta revisión 
1. Sobre los errores indicados, y no borrando todos los comentarios y volviendo a poner el mismo contenido de nuevo con justificaciones diferentes
2. Tratando de justificar todos los cambios como solución de problemas, *como se ha hecho desde el principio*
3. Tratando de entender lo que se busca en la asignatura en general y en este objetivo en particular.

Por todo esto también voy a esperar que las dos personas que han revisado hasta ahora (y a las que agradezco su trabajo, que será recompensado con la nota correspondiente) vuelvan a revisar lo que se envíe y que no se me pida revisión hasta que ellas lo hayan aprobado.

JJ commented:
Te recuerdo también que *todos los commits*, que son cambios en el código, deben aportar valor y resolver un problema. Borrar varios ficheros sin justificar con un issue correspondiente tampoco es una buena práctica, y ya es la tercera vez que lo haces.

gabrielherreraloz commented:
> Has vuelto a eliminar el Dockerfile para eliminar los comentarios por tercera vez. No es una buena idea, y te rogaría que no lo volvieras a hacer. Las dos veces anteriores no han resultado en un acercamiento al objetivo sino, al contrario, has vuesto a enviar el mismo objetivo con justificaciones diferentes, lo que es un error. Te rogaría que hicieras cambios _puntuales_ en las líneas indicadas. Te recuerdo también que a partir de la tercera revisión (y llevas ya 4) solo se hacen revisiones adicionales si hay alguna mejora; si no hay mejora se puede fallar el objetivo en la convocatoria en la que se esté enviando, que es la ordinaria. Así que te rogaría que hicieras esta revisión
> 
> 1. Sobre los errores indicados, y no borrando todos los comentarios y volviendo a poner el mismo contenido de nuevo con justificaciones diferentes
> 2. Tratando de justificar todos los cambios como solución de problemas, _como se ha hecho desde el principio_
> 3. Tratando de entender lo que se busca en la asignatura en general y en este objetivo en particular.
> 
> Por todo esto también voy a esperar que las dos personas que han revisado hasta ahora (y a las que agradezco su trabajo, que será recompensado con la nota correspondiente) vuelvan a revisar lo que se envíe y que no se me pida revisión hasta que ellas lo hayan aprobado.

¿Cómo que "para eliminar los comentarios"? No lo hago para eliminar los comentarios, lo hago precisamente porque todo el código que tengo y documentación no aporta nada de valor y mi intención es empezar desde 0 utilizando multi-stage, que es lo que planteé originalmente aunque finalmente cambiara de idea, porque sí, me informé sobre ello bastante. De hecho, he borrado el código y he realizado un squash de los commits puesto que precisamente eso comentamos en clase si lo que íbamos a hacer era empezar desde cero, ya que también voy a replantear uno de los issues sobre el que están hechos muchos commits, si elimino ese issue directamente, los commits asociados no valdrán para nada, por eso borro el código. Todo ellos lo hago viendo la reiteración de los comentarios sobre el código, y asumiendo que tengo un mal planteamiento base de todo lo realizado hasta ahora en el objetivo, y con la intención de intentar retomar la buena dinámica adquirida al final del periodo docente en diciembre, el cual he perdido por dedicar tanto tiempo a las demás asignaturas que tengo. Gracias por los consejos.

JJ commented:
> > Has vuelto a eliminar el Dockerfile para eliminar los comentarios por tercera vez. No es una buena idea, y te rogaría que no lo volvieras a hacer. Las dos veces anteriores no han resultado en un acercamiento al objetivo sino, al contrario, has vuesto a enviar el mismo objetivo con justificaciones diferentes, lo que es un error. Te rogaría que hicieras cambios _puntuales_ en las líneas indicadas. Te recuerdo también que a partir de la tercera revisión (y llevas ya 4) solo se hacen revisiones adicionales si hay alguna mejora; si no hay mejora se puede fallar el objetivo en la convocatoria en la que se esté enviando, que es la ordinaria. Así que te rogaría que hicieras esta revisión
> > 
> > 1. Sobre los errores indicados, y no borrando todos los comentarios y volviendo a poner el mismo contenido de nuevo con justificaciones diferentes
> > 2. Tratando de justificar todos los cambios como solución de problemas, _como se ha hecho desde el principio_
> > 3. Tratando de entender lo que se busca en la asignatura en general y en este objetivo en particular.
> > 
> > Por todo esto también voy a esperar que las dos personas que han revisado hasta ahora (y a las que agradezco su trabajo, que será recompensado con la nota correspondiente) vuelvan a revisar lo que se envíe y que no se me pida revisión hasta que ellas lo hayan aprobado.
> 
> ¿Cómo que "para eliminar los comentarios"? No lo hago para eliminar los comentarios, lo hago precisamente porque todo el código que tengo y documentación no aporta nada de valor y mi intención es empezar desde 0 utilizando multi-stage, que es lo que planteé originalmente aunque finalmente cambiara de idea, porque sí, me informé sobre ello bastante. De hecho, he borrado el código y he realizado un squash de los commits puesto que precisamente eso comentamos en clase si lo que íbamos a hacer era empezar desde cero, ya que también voy a replantear uno de los issues sobre el que están hechos muchos commits, si elimino ese issue directamente, los commits asociados no valdrán para nada, por eso borro el código. Todo ellos lo hago viendo la reiteración de los comentarios sobre el código, y asumiendo que tengo un mal planteamiento base de todo lo realizado hasta ahora en el objetivo, y con la intención de intentar retomar la buena dinámica adquirida al final del periodo docente en diciembre, el cual he perdido por dedicar tanto tiempo a las demás asignaturas que tengo. Gracias por los consejos.

No sé por qué lo haces, Gabriel. El hecho es que, como te digo [aquí](https://github.com/gabrielherreraloz/IV-GHL/pull/26#issuecomment-3737643346), debes hacer commits que aporten valor y que resuelvan problemas. Si lo que quieres es usar multi-stage, **también tienes que poner qué problema resuelve**, que no es ninguno de los que se te han indicado hasta el momento, y por supuesto *tratar de hacer cambios incrementales sobre lo que hay*, que para eso son los pull requests.

Si quieres retomar la dinámica del periodo lectivo, trata de hacer lo que se te ha dicho desde el principio
1. Enunciado claro de los problemas en issues
2. Cambios puntuales *sobre lo ya hecho* y siguiendo los comentarios
3. Siempre seguir las buenas prácticas con commits atómicos y mensajes de commit que expliquen decisiones. ya te dije [aquí](https://github.com/gabrielherreraloz/IV-GHL/pull/26#issuecomment-3701990152) que *tenías que abrir issues para indicar los problemas* y eso es lo que se ha dicho desde el principio en esta asignatura. Y esa era la primera revisión. De veras que siento que hayas tenido que detraer tiempo de otras asignaturas, pero si no sigues los consejos que se te dan vas a caer de nuevo en los mismos errores. 

Nadie te ha pedido que hagas un Dockerfile en dos fases. Tu Dockerfile tiene problemas específicos, que se te han señalado repetidamente. ¿Borrarlo y hacerlo de nuevo soluciona esos problemas? Haces cambios extremos y sin justificar para cosas concretas que podías haber resuelto simplemente borrándolas, como cuando [se te dijo que no se podían copiar los ficheros del volumen montado al contenedor](https://github.com/gabrielherreraloz/IV-GHL/pull/26#discussion_r2653511605) no abriste un issue y eliminaste esos comandos, borraste el Dockerfile (y como has hecho force push varias veces esos commits también han desaparecido). Cuando se te dijo [aquí](https://github.com/gabrielherreraloz/IV-GHL/pull/26#discussion_r2653509445) si los privilegios de grupo eran necesarios, primero, no abriste un issue para indicar realmente los privilegios que eran necesarios; dejaste los privilegios de grupo ¡y además le añadiste el sticky bit!

Si hubieras abierto un issue *en ese momento* podrías haber dejado ese mismo `777` simplemente justificando que puede haberse creado un usuario en el mismo grupo que el que existe (por casualidad) o cambiarlo a 757 indicando que el usuario tiene su propio grupo (menos probable, pero plausible) y que no va a haber más usuarios en el mismo. Sin embargo, lo has cambiado a `1777` que *no resuelve ese problema inicial* porque no justifica que estés entendiendo lo que estés haciendo ni por qué y además lo has vuelto a dejar *a través de dos iteraciones*

De veras que no entiendo por qué haces esto mucho más complicado que lo que es. Las revisiones del PR están hechas para que el estudiante entienda qué es lo que debe comprender para poder progresar hacia la consecución del objetivo. En vez de usarlas, revisas *contra* ellas alejándote cada vez más de la consecución del mismo, con lo que en cada iteración se ven más fallos que no se veían al principio (por ejemplo, lo que te he comentado sobre Alpine para Go no lo habría visto si no me hubiera puesto a leer el README de la imagen de Go).

Eres libre por supuesto de hacer lo que quieras y borrar todos los ficheros para comenzar desde el principio. Pero si no resuelves los problemas que se te han indicado en las revisiones del principio, ni realmente entiendes *por qué* tienes que resolverlos, no es ni para ti ni para mi la mejor forma de emplear ese tiempo que le estás restando a tus otras asignaturas.

JJ commented:
Por favor, tratad de terminar esto ya, y en todo caso superarlo mañana a más tardar. Aunque la guía docente da un día con respecto a la fecha final de evaluación, son dos días no lectivos y por tanto el último día hábil sería mañana.

gabrielherreraloz commented:
> Por favor, tratad de terminar esto ya, y en todo caso superarlo mañana a más tardar. Aunque la guía docente da un día con respecto a la fecha final de evaluación, son dos días no lectivos y por tanto el último día hábil sería mañana.

Me es imposible cumplir la fecha para la ordinaria, así que cuanto con entregarlo ya oara la extraordinaria.

gabrielherreraloz committed:
Creada documentación imagen docker(#27)

Se ha documentado el prodceso de selección de la imagen docker base para el dockerfile, indicando los criterios de selección y analizando el por qué del descarte de cada opción no seleccionada, y citando todas las fuentes de las que se ha extraido información.

gabrielherreraloz committed:
Creado Dockerfile básico(#27)

Se ha implementado un archivo Dockerfile utilizando bitnami como imagen base ya que así se ha justificado en la documentación.
Posteriormente, ejecutando "RUN install_packages make" se instala la herramienta make, necesaria para lñanzar los tests, siendo install_packages una orden especial de bitnami que automaticamente hace update, intall y clean, ahorrando código.
Por último, se selecciona como directorio principal WORKDIR como se especifica en la documentación del objetivo, y se lanzan los tests utilizando make, ya que es nuestro gestor de tareas.

gabrielherreraloz committed:
Ejecutando tests con usuario no priv.(#30)

A pesar de que por defecto, Bitnami ejecuta todo el dockerfile utilizando el usuario sin privilegios 1001, nos aseguramos que esto ocurra indicandolo antes de especificar el directorio principal.

gabrielherreraloz committed:
Permisos de cache a Go(#31)

Para la correcta ejecución de tests, go necesita un directorio sobre el que escribir la cache, por ello creamos un directorio específico para esto y le otorgamos permisos de lectura y escritura para que go pueda ejecutar correctamente.
En principio, como se ha creado en ese mismo momento el directorio, no hace falta cambiar a root para modificar los permisos de este directorio.

gabrielherreraloz commented:
@GabrielFranciscoSM si me pudieras revisar por favor, he modificado por completo la documentación y el dockerfile, así como he planteado nuevos issues mas específicos.

gabrielherreraloz commented:
> @GabrielFranciscoSM si me pudieras revisar por favor, he modificado por completo la documentación y el dockerfile, así como he planteado nuevos issues mas específicos.

@jorgelopez-ugr si puedes tu también porfa

jorgelopez-ugr commented on Dockerfile:
```diff
@@ -0,0 +1,12 @@
+
+RUN install_packages make
+
+RUN mkdir -p /tmp/go-cache && chmod 777 /tmp/go-cache
```
Piensa que usuario está ejecutando estas operaciones ahora mismo. Cuando sepas cual es, piensa en si eso es seguro.

gabrielherreraloz committed:
Cambiada línea de usuario sin priv.(#30)

Se ha cambiado la seleccion del usuario sin privilegios a antes de lña asignación de la crpeta cache para mayor seguridad, siguiendo el comentario de @jorgelopez-ugr.

jorgelopez-ugr reviewed (commented):
Por el momento échale un vistazo al dockerfile que hay cosillas.

GabrielFranciscoSM commented on docs/Contenedores.md:
```diff
@@ -0,0 +1,38 @@
+  Es la última versión estable de Debian, incluye la herramienta Make instalada por defecto, sin embargo, su uso del almacenamiento es mucho mayor, ocupando aproximadamente unos 300MB de forma comprimida, y al expandirse en el disco llega a pesar 1,2GB.
+
+  **Oficial: Basada en Alpine**: [Repositorio](https://github.com/docker-library/golang/blob/adf0ff9a10a766d7dc1b394d52544f3a1b1da4e2/1.25/alpine3.23/Dockerfile)
+  Cuenta con un tamaño mucho menor que la versión de Debian, pesando aproximadamente unos 64,5MB de forma comprimida, llegando a pesar 329MB al expandirse en el disco, por otro lado, no incluye Make instalada por defecto, aunque puede instalarse en un paso posterior sin apenas aumentar el uso de almacenamiento, ni el tiempo de lanzamiento del contenedor. Sin embargo, ejn la documentación oficial  ...
```
No veo que no elijas esta imagen por las razones expuestas, pero ¿No te estás saliendo de los criterios expuestos anteriormente?

gabrielherreraloz replied:
Tienes razón, voy a exponer mejor los criterios

GabrielFranciscoSM commented on docs/Contenedores.md:
```diff
@@ -0,0 +1,38 @@
+  Cuenta con un tamaño mucho menor que la versión de Debian, pesando aproximadamente unos 64,5MB de forma comprimida, llegando a pesar 329MB al expandirse en el disco, por otro lado, no incluye Make instalada por defecto, aunque puede instalarse en un paso posterior sin apenas aumentar el uso de almacenamiento, ni el tiempo de lanzamiento del contenedor. Sin embargo, ejn la documentación oficial  ...
+
+  **Oficial: Scratch**: [Documentación](https://hub.docker.com/_/scratch)
+  Se trata de una imagen vacía sin nada instalado, es la opción más ligera y segura, sería una excelente opción en caso de usar la técnica de multi-stage, anteriormente mencionada, pero en este caso, una imagen tan vacía no es la mejor opción.
```
¿Por qué no es la mejor opción?

GabrielFranciscoSM commented on docs/Contenedores.md:
```diff
@@ -0,0 +1,38 @@
+  Es una opción muy parecida a Scrath pero algo menos vacía, tiene un peso de 830kB de manera comprimida, con un peso final de 6,23MB. Es ultrasegura y muy recomendada para entornos que utilizan multistage, pero en este caso no es la opción más recomendad por los mismos motivos que se han comentado con Scratch.
+
+  **No Oficial: Bitnami**: [Documentación](https://hub.docker.com/r/bitnami/golang)
+  Se trata de una opción no oficial pero mantenida por la ya conocida empresa VMware, lo que aporta cierta fiabilidad a esta imagen a pesar de ello. Se trata de una versión reducida de debian y muy segura, ya que no utiliza root por defecto. Al igual que alpine, no trae make instalado por defecto pero se puede instalar de forma manual, e incluso de manera más simplificada que en la imagen estanda ...
```
¿El renombre de la empresa mantenedora?¿Es la seguridad un criterio de comparación?

gabrielherreraloz replied:
Lo he puesto como info adicional, pero ahora que lo pienso puedo poner que si es una imagen no oficial debe ser de una fuente fiable.

GabrielFranciscoSM replied:
Creo que ha quedado un poco borde la primera pregunta, pero no era la intención, solo que me quedé a medias.

gabrielherreraloz replied:
Nah estaba perfe no te preocupes

GabrielFranciscoSM reviewed (commented):
Te he comentado sobre la elección de las imágenes.

gabrielherreraloz committed:
Mejorados criterios documentación(#29)

Se han añadido más criterios con el fín de esclarecer la explicación y justificación de la imagen elegida, así como del descarte de las demás.

GabrielFranciscoSM reviewed (approved):
lo veo bien

gabrielherreraloz committed:
Eliminando comentarios residuales

Se ha modificado la linea de los comentarios que se habian resuelto pero no había cerrado el propio comentario.

jorgelopez-ugr reviewed (approved):
Buen trabajo bien revisado

gabrielherreraloz commented:
@JJ listo

JJ commented on .github/workflows/docker-publish.yml:
```diff
@@ -0,0 +1,33 @@
+on:
+  push:
+    branches: [ "main" ]
+    tags: [ 'v*.*.*' ]
```
No me pongáis cosas raras, hombre...

gabrielherreraloz replied:
No lo había puesto yo, estaba en la base de la github action del propio github, se que es para el etioquetado de versiones pero se me pasaría borrarlo, lo quito.

JJ replied:
En general, conviene que no pongáis cosas que no sabéis que son; aunque no es el tema de este objetivo, siempre viene bien darle una revisión, a esto y a las versiones de las actions, que casi nunca pone las últimas. Simplemente por no usar código que no se conozca bien qué es lo que se hace.

JJ commented on docs/Contenedores.md:
```diff
@@ -0,0 +1,42 @@
+  Se trata de una imagen vacía sin nada instalado, es la opción más ligera y segura, sería una excelente opción en caso de usar la técnica de multi-stage, anteriormente mencionada, pero en este caso, una imagen tan vacía no es la mejor opción, ya que al no incluir make, scratch no tiene ninguna forma de instalarlo pues solo trabaja con binarios directamente, lo cual descarta esta opción.
+
+  **No Oficial: GoogleDistroless**: [Documentación](https://github.com/GoogleContainerTools/distroless)
+  Es una opción muy parecida a Scrath pero algo menos vacía, tiene un peso de 830kB de manera comprimida, con un peso final de 6,23MB. Es ultrasegura y muy recomendada para entornos que utilizan multistage, pero en este caso no es la opción más recomendada por los mismos motivos que se han comentado con Scratch.
```
Typo

JJ commented on Makefile:
```diff
@@ -2,4 +2,7 @@ install:
 	go mod tidy
 
 check:
-	go fmt ./...
```
No sé por qué has cambiado esto, pero siempre os he dicho lo mismo sobre cambios no relacionados con el objetivo.

gabrielherreraloz replied:
Lo he cambiado porque los tests no se estaban lanzando a traves de make, lo cual no tenia mucho sentido si lo tengo como gestor de tareas, entonces le he añadido que lance los test

gabrielherreraloz replied:
> Lo he cambiado porque los tests no se estaban lanzando a traves de make, lo cual no tenia mucho sentido si lo tengo como gestor de tareas, entonces le he añadido que lance los test

Lo he cambiado en este objetivo porque como va relacionado con lanzar los tests a través de docker...

JJ replied:
OK, que solo has cambiado el retorno de carro...
Generalmente es buena costumbre terminar todos los ficheros en un retorno de carro para que cuando se añadan líneas nuevas no aparezca la última línea como modificada.

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,12 @@
+
+RUN install_packages make
+
+USER 1001
```
¿Por qué usas esto y no el nombre del usuario?
Aparte de que *no* debéis crear ningún usuario con ningún UID específico. Y este ni siquiera está creado. [Aquí](https://github.com/gabrielherreraloz/IV-GHL/pull/26#discussion_r2681041067) te hablé explícitamente del grupo. No te dije nada del usuario. Has cambiado toda esa línea por otra que no tiene sentido. la has hecho como respuesta a #30 que dice que no puede ser root, *pero no dice que tenga que ser el usuario 1001*, que además ni siquiera está creado. ¿Has mirado si la imagen de bitnami tiene algún usuario definido?

gabrielherreraloz replied:
Si, lo he mirado, el usuario 1001 es el usuario por defecto en bitnami, por ello no lo creo y lo uso directamente, pensé en crear uno pero decidí aprovechar el que ya venía por defecto, de hecho bitnami, ejecuta todo en no-root de forma predeterminada.

JJ replied:
Ese usuario tendrá un nombre... En todo caso tratándose del 1001 deberías crear otro usuario no privilegiado, con un valor por defecto. El valor 1001 es el único que no debes usar, simplemente porque va a hacer que el test funcione porque es con el mismo usuario con el que se lanza, que es lo que *no* queremos.
Y por supuesto que deberías haberlo justificado en el mensaje de commit y no lo has hecho. Aparte de mostrarme el usuario definido en ese momento o ahora.

JJ replied:
Y sinceramente, no sé dónde has visto eso en la documentación. Me he metido en la definición del Dockerfile y no he visto absolutamente ningún usuario definido.

gabrielherreraloz replied:
> Y sinceramente, no sé dónde has visto eso en la documentación. Me he metido en la definición del Dockerfile y no he visto absolutamente ningún usuario definido.

No sale en la documentación pero se comenta en varios sitios:
https://stackoverflow.com/questions/61826463/setting-the-user-uid-in-a-bitnami-docker-container
https://github.com/bitnami/containers/issues/65827

No se si lo he podido interpretar mal, pero lo cambio igualmente.

JJ reviewed (commented):
Sigues borrando líneas donde se señala un error sin arreglar ese error explícitamente y añadiendo nuevas líneas que tienen errores diferentes. Por favor, tratad de hacer mínimamente lo que se os corrige, porque si no es imposible que avancéis.

gabrielherreraloz committed:
Eliminado etiquetado versiones(#27)

Se ha eliminado la restricción sobre el etiquetado de las versiones de la imagen de docker ya que no es necesario para este objetivo.

gabrielherreraloz committed:
Cambiado usuario de ejecución(#30)

Se ha cambiado la utilización del usario 1001, creado por defecto en bitnami, por la creación de un usuario con nombre específico por mí.
Se ha tomado esta decisión ya que se testeará utilizando el usuario 1001, y utilizando un usuario diferente en el dockerfile, nos aseguramos que funciona siempre, y no solo por estar utilizando el mismo usuario.

gabrielherreraloz committed:
FIX- Errata(#29)

Se había escrito Scratch como Scrath, se ha corregido.

gabrielherreraloz committed:
Eliminado comentario resuelto

gabrielherreraloz commented:
@JJ Vuelvo a pedir revisión a mis compañeros?

JJ commented:
No hace falta

JJ reviewed (approved):
👍
